Assets/Scripts/Weapons/Weapon.cs

```diff
@@ -0,0 +1,12 @@
+﻿using UnityEngine;
+
+namespace Weapons
+{
+    [RequireComponent(typeof(SpriteRenderer))]
+    public abstract class Weapon : MonoBehaviour
+    {
+        public SpriteRenderer spriteRenderer;
```

Comment:
Wenn du RequireComponent hast, kannst du eigentlich auch GetComponent in Awake machen:P